core/themes/bartik/templates/comment.tpl.php

```diff
@@ -18,19 +18,22 @@
  *   desired parameters on the $comment->changed variable.
  * - $new: New comment marker.
  * - $permalink: Comment permalink.
+ * - $permalink_path: Available to build your own permalink.
  * - $submitted: Submission information created from $author and $created during
  *   template_preprocess_comment().
  * - $picture: Authors picture.
  * - $signature: Authors signature.
  * - $status: Comment status. Possible values are:
  *   comment-unpublished, comment-published or comment-preview.
  * - $title: Linked title.
+ * - $title_options: Auto-generated(0), custom(1) or hidden(2).
```

Comment:
Already better, no more constants. But what I actually had in mind: provide something more "speaking"

"title_display":

- generated / custom = TRUE
- hidden = FALSE

Reply:
I see, I need to be more clear about my suggestion. :wink:

It's about how it is used in the template (hence display). For that we only need some sort of on/off switch to either render the title (be it generated or custom), or not render the title (when it's hidden). So a single on/off thing seems sufficient for this particular usage in comment.tpl.php. The wrapper css class allows finer grained theming.

My suggestion is really only about the render switch in the preprocessor for the template, I'm fine with the three radios in the setting, which also control if the title form item is provided or not.

What may be a bit confusing is, that the radios (and constants) technically control two different things:

- The form item (textfield) in the comment form
- The title rendering when displaying comments

But I still think, that's a good idea - and that's what we settled on. In fact, form item and display rendering *are* related (as discussed). If we allow people to set a custom comment title (subject), we for sure don't want to hide it in display. If we generate it automatically, we might want to hide it to prevent that odd duplicate text.

Reply:
I just realized, I commented on the wrong tpl.php file. This should have gone to the main one in core/modules/comment/templates/comment.tpl.php. :wink:

Reply:
@indigoxela - I THINK I get your point now. Thanks for your patience. 

If I understand you correctly, the change you are requesting does not functionally change the way the PR works but is more direct/understandable in the comment.tpl.php file. I think you are suggesting that instead of passing the original variable which handles more than just the title decision, I only pass need to pass a TRUE/FALSE varaible to the template regarding whether titles display or not. 

This will function the same, but will make more sense and be easier to understand for themers. 

I like this idea, which would not have occurred to me on my own. I've made that change. If I'm still missing any part of your suggestion, let me know. I also welcome comments on my implementation.

Reply:
I'm not sure either variable is necessary. Can't we just check if the title itself is empty or not? If the title is set to hidden, then the `$title` is an empty string, isn't it? Why do we need another variable?